src/resmom/mock_run.c

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (C) 1994-2020 Altair Engineering, Inc.
+ * For more information, contact Altair at www.altair.com.
+ *
+ * This file is part of the PBS Professional ("PBS Pro") software.
+ *
+ * Open Source License Information:
+ *
+ * PBS Pro is free software. You can redistribute it and/or modify it under the
+ * terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation, either version 3 of the License, or (at your option) any
+ * later version.
+ *
+ * PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Commercial License Information:
+ *
+ * For a copy of the commercial license terms and conditions,
+ * go to: (http://www.pbspro.com/UserArea/agreement.html)
+ * or contact the Altair Legal Department.
+ *
+ * Altair’s dual-license business model allows companies, individuals, and
+ * organizations to create proprietary derivative works of PBS Pro and
+ * distribute them - whether embedded or bundled with other software -
+ * under a commercial license agreement.
+ *
+ * Use of Altair’s trademarks, including but not limited to "PBS™",
+ * "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+ * trademark licensing policies.
+ *
+ */
+
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <time.h>
+
+#include "attribute.h"
+#include "job.h"
+#include "mock_run.h"
+#include "mom_func.h"
+#include "pbs_error.h"
+#include "resource.h"
+
+
+/**
+ * @brief	work task handler for end of a job in mock run mode
+ *
+ * @param[in]	ptask - pointer to the work task
+ *
+ * @return void
+ */
+void
+mock_run_end_job_task(struct work_task *ptask)
+{
+	job *pjob;
+
+	if (ptask == NULL)
+		return;
+
+	pjob = ptask->wt_parm1;
```

Comment:
@agrawalravi90 I guess I was not clear, I meant that if a job is deleted then ptask->wt_parm1 becomes a dangling job pointer, isn't it? In such a case, dereferencing it would result into undesirable outcome (like crash etc.).
Now I think to address it, either we can just store the job id in param1 and do a find_job on the job id in this function. But this will be a little more time consuming because of the find.
OR
We can just delete the task (see delete_task_by_param1_func()) using param1 (which is job pointer) whenever we delete a job in "mock_run" mode.
OR
We just don't care about any of these things because it in testing mode .

I personally would prefer second option. Let me know what you think.

Thanks!

Reply:
thanks for clarifying. i tried a qdel and it got stuck, so I need to investigate the delete further, marking the PR as WIP for now.